### PR TITLE
Unit Tests: Add comprehensive coverage for `BlockEditorVarianceHandler`

### PR DESCRIPTION
**Pull request by kiannidev**
## Summary

Adds test coverage for `BlockEditorVarianceHandler` and refactors existing tests to reduce duplication. This PR contains **test-only changes** - no production code modifications.

## Changes from Main Branch

**Test Coverage:**
- Added **6 new test methods**:
  - `Assigns_Default_Culture_When_Culture_Variance_Is_Enabled`
  - `Removes_Default_Culture_When_Culture_Variance_Is_Disabled`
  - `Ignores_NonDefault_Culture_When_Culture_Variance_Is_Disabled`
  - `AlignPropertyVarianceAsync_Removes_NonDefault_Culture_Values`
  - `AlignExposeVariance_Skips_When_ElementType_Not_Found`
  - `AlignExposeVariance_Handles_Multiple_ContentData_Items`

**Code Quality Improvements:**
- Created helper methods to eliminate code duplication:
  - `CreateBlockListValue` (with overloads)
  - `CreateBlockPropertyValues`
  - `CreateBlockItemVariations`
  - `ExecuteAlignExposeVariance`
  - `ExecuteAlignPropertyVarianceAsync`
  - `ExecuteAlignedPropertyVarianceAsync`
  - `SetupAlignedExposeTest`
- Introduced `BlockPropertyValueConfig` record class to reduce argument count
- Refactored all existing tests to use helper methods consistently

## Testing

All tests pass. No production code changes.

### CODE REVIEW

**Comment by AndyButland** [comment]
Are you still working on this PR please @kiannidev or do you consider it ready for review?  If the latter, please could you go over and update the description so it's clear what you are aiming to do here.  You mention 8 new tests but I only see 2 and not the refactoring you describe, plus what seems to have started as purely test coverage has also led to some changes to production behaviour.  If you can give a good description of what you are looking to do, that will help with reviewing. Thanks.

**Comment by kiannidev** [comment]
Hi, @AndyButland 
Sorry for confusion.
I added 8 Tests first, but there was a lint error about repeatation, so I tried to reduce test cases, but I didn't update description.
I will add some tests again and update description.
Sorry for that.

Thanks

**Comment by kiannidev** [comment]
Hi, @AndyButland 
I added more test cases and updated description to reflect changes properly.
Sorry for confusion.
Now this is ready to review.

Thanks

**Comment by AndyButland** [comment]
Thanks @kiannidev.  If you can, I'd like to understand the two bugs you found.  How do these manifest themselves in practice? I'd like to do a manual verification too, where I can see the issue before your changes and verify it's fixed afterwards.

**Comment by kiannidev** [comment]
**Bug 1: Deduplication Not Always Running**
Logic error: Deduplication only ran when alignment was needed. If entries were already aligned, an early return skipped deduplication, leaving duplicates.
Fix: Deduplication now always runs at the end (line 219).
**Bug 2: Orphaned Expose Entries Not Removed**
Logic error: When blocks are deleted from ContentData, their Expose entries weren't removed, leaving orphaned references.
Fix: Added cleanup to remove expose entries that don't match any ContentData keys (lines 167, 202).
Both are logic errors (bugs in behavior), not compilation errors. The code compiled and ran, but behaved incorrectly.

File: tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs [hunk 1]
```diff
@@ -58,88 +51,276 @@ public void AlignExpose_Can_Align_Invariance()
...
-
-        Assert.Multiple(() =>
-        {
+        var values = CreateBlockPropertyValues(("one", "en-US", "segment-one", "Value one"));
+        var expose = CreateBlockItemVariations((contentDataKey, null, null));
+        var blockValue = CreateBlockListValue(contentDataKey, owner.ContentType.Key, values, expose);
+        ExecuteAlignExposeVariance(owner, blockValue);
             var alignedExpose = blockValue.Expose.First();
             Assert.AreEqual("en-US", alignedExpose.Culture);
             Assert.AreEqual("segment-one", alignedExpose.Segment);
```
Comment (AndyButland) [inline tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs:216]: An `Assert.Multiple` block has been removed here. I'd be happy to keep that, but if removed, at least the indentation should be cleaned up.

File: tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs [hunk 2]
```diff
@@ -58,88 +51,276 @@ public void AlignExpose_Can_Align_Invariance()
...
+    {
+        var owner = PublishedElement(ownerVariation);
+        var subject = BlockEditorVarianceHandler("da-DK", owner);
+        return await subject.AlignedPropertyVarianceAsync(
+            propertyValue,
+            PublishedPropertyType(propertyTypeVariation),
+            owner);
+    }
+
+    private static BlockListValue CreateBlockListValue(Guid contentDataKey, Guid contentTypeKey, List<BlockPropertyValue> values, List<BlockItemVariation> expose)
```
Comment (AndyButland) [inline tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs]: `CreateBlockListValue` is a good addition, being handy helper to DRY up some of the test setup, but it's not used in all tests.  Could you look at consistently using it?

File: tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs [hunk 3]
```diff
@@ -58,88 +51,276 @@ public void AlignExpose_Can_Align_Invariance()
...
+        var blockValue = CreateBlockListValue(contentDataKey, owner.ContentType.Key, values, expose);
+        ExecuteAlignExposeVariance(owner, blockValue);
             var alignedExpose = blockValue.Expose.First();
             Assert.AreEqual("en-US", alignedExpose.Culture);
             Assert.AreEqual("segment-one", alignedExpose.Segment);
-        });
     }
 
     [Test]
-    public void AlignExpose_Can_Handle_Variant_Element_Type_With_All_Invariant_Block_Values()
```
Comment (AndyButland) [inline tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs]: The test `AlignExpose_Can_Handle_Variant_Element_Type_With_All_Invariant_Block_Values` seems to have been removed, and I don't see an equivalent replacement.  It would be preferred to bring this back.

File: tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs [hunk 4]
```diff
@@ -167,7 +355,14 @@ private static BlockEditorVarianceHandler BlockEditorVarianceHandler(string defa
...
         elementType.SetupGet(e => e.Key).Returns(element.ContentType.Key);
         elementType.SetupGet(e => e.Variations).Returns(element.ContentType.Variations);
-        contentTypeServiceMock.Setup(c => c.Get(element.ContentType.Key)).Returns(elementType.Object);
+        contentTypeServiceMock.Setup(c => c.Get(It.IsAny<Guid>())).Returns((Guid key) =>
+        {
+            if (key == element.ContentType.Key)
+            {
+                return elementType.Object;
+            }
+            return null!; // Return null for unknown content types
```
Comment (AndyButland) [inline tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs]: I haven't tested yet to see why this is needed, but it looks a little odd (given we are in test code, would we expect unknown object types?).  Maybe you could re-look at this and justify in a reply to this comment why it's necessary.

**Comment by kiannidev** [comment]
> Looks a good addition to test coverage, thanks @kiannidev. Other than the inline comments, could I ask you to please move the two production fixes into separate PRs, with linked issues, that show reproduction steps and/or targeted tests that prove the bug exists. Otherwise I feel mixing behaviour changes into a "test coverage" PR makes review harder and the git history less clear, so I'd prefer if we can separate this out and keep this update purely related to testing. We avoid the any risk then of introducing an inadvertent regression that might be hard to find when included with a PR that's not strictly related to fixing a bug.

Hi, @AndyButland 
You mean that I should create new Issues?
And separate this PR into 2 PRs what is linked those issues?

**Comment by kiannidev** [comment]
Hi, @AndyButland 
I created 2 issues to reflect the bugs and created 1 PR to resolve the 2 issues what I created.
Please merge it first, so that I can update this PR to include only adding Tests.
If you don't merge the PR 21801 and I make this PR include only adding Tests, some lint errors are occured.


Thanks

File: tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs (hunk 1 above)
Comment (kiannidev) [reply tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs:216]: Thanks for the feedback! I've kept the `Assert.Multiple` block as you suggested. The indentation has been verified and is correct - the code inside the block is properly indented. The `Assert.Multiple` block is now at lines 74-79 and all assertions inside are correctly nested.

File: tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs (hunk 2 above)
Comment (kiannidev) [reply tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs]: I've updated all tests to consistently use the CreateBlockListValue helper method. I've also added an overload to handle cases with multiple ContentData items. All manual BlockListValue creations have been replaced with the helper method, ensuring consistent test setup across all tests.

File: tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs (hunk 3 above)
Comment (kiannidev) [reply tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs]: I've restored the AlignExpose_Can_Handle_Variant_Element_Type_With_All_Invariant_Block_Values test.

File: tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs (hunk 4 above)
Comment (kiannidev) [reply tests/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/PropertyEditors/BlockEditorVarianceHandlerTests.cs]: The null! return is necessary because it simulates real-world behavior where IContentTypeService.Get() can return null for non-existent content types. The production code in BlockEditorVarianceHandler.cs (line 171-172) handles this by calling _contentTypeService.Get() and then using .WhereNotNull() to filter out null results. This means the production code is designed to gracefully handle cases where content types aren't found.

The test AlignExposeVariance_Skips_When_ElementType_Not_Found specifically verifies this behavior - it uses an unknownContentTypeKey that doesn't match any registered content type, and verifies that the code correctly skips processing that content data when the content type lookup returns null.

I've updated the comment to make this clearer. The null return is intentional and necessary to test this edge case properly.

**Comment by kiannidev** [comment]
Hi, @AndyButland 
Thanks for merging the PR 21801.
I separated bug fix part into that PR and let this PR contain only other tests and resolved inline comments.
Please review them and give me any feedbacks...

Thanks
